feat(universes): Add place-level user restriction support

commonly-ts opened the pull request:
# Description

<!-- Provide a clear and concise description of your changes -->

Extends the Universes class to allow updating user restrictions at the place level by accepting an optional placeId parameter. Adjusts the resource path logic to handle both universe-level and place-level user restrictions.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] ✅ Test update

## Changes Made

<!-- List the specific changes made in this PR -->

- Adds `placeId` parameter to `universes.updateUserRestriction`
- Updates `universes.updateUserRestriction` to use an options object for `universeId`, `placeId` and `body`.
- Adds placeId check to the updateUserRestriction test.

## Testing

<!-- Describe the tests you ran to verify your changes -->

- [x] All existing tests pass (`pnpm test`)
- [ ] Added new tests for new functionality
- [ ] Tested manually (describe below)

### Manual Testing

Simple string concatenation so no manual testing needed

## Code Quality

<!-- Confirm you've followed the project's standards -->

- [x] My code follows the project's coding style
- [x] I have run `pnpm lint` and fixed any issues
- [x] I have run `pnpm format` to format my code
- [x] I have run `pnpm typecheck` and there are no type errors
- [x] I have reviewed my own code

## Documentation

<!-- Ensure documentation is up to date -->

- [ ] I have updated the relevant documentation (if applicable)
- [x] I have added/updated JSDoc comments for new/modified code

## Screenshots/Recordings

<img width="760" height="440" alt="image" src="https://github.com/user-attachments/assets/49e5bb90-d8df-43d9-b34f-2147aaa09e86" />

## Checklist

<!-- Final checks before submitting -->

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guide
- [x] My PR title follows the [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] I have rebased my branch on the latest `main` branch
- [ ] I have tested my changes against the Roblox Open Cloud API (if applicable)
- [x] This PR is ready for review

---

<!--
Thank you for contributing to @relatiohq/opencloud! 🎉
Your contribution helps make this SDK better for everyone.
-->

commonly-ts committed:
Add place-level user restriction support

Extends the Universes class to allow updating user restrictions at the place level by accepting an optional placeId parameter. Adjusts the resource path logic to handle both universe-level and place-level user restrictions.

commonly-ts commented:
I would agree if the method's parameters are a bit icky now, I'm fine with changing it into options typed with an interface if you'd like

marinofranz reviewed (changes requested):
I agree, let's make this into an options object so we can re-use
There are a couple of other methods that have this feature as well, you can integrate the options object into that as well if you're feeling up for it 🙂 

Make sure to add this to a test or create a new one so coverage can pass

commonly-ts committed:
Added options to updateUserRestriction

commonly-ts committed:
Updated updateUserRestriction test

commonly-ts committed:
fix: handle empty duration in user restriction update

Prevents API errors by setting 'duration' to undefined if it is an empty string when updating user restrictions. This avoids sending invalid data that would result in a 400 response from the API.

commonly-ts committed:
fix: linting

commonly-ts committed:
fix: handle empty duration in user restriction updates

Refactored Universes.updateUserRestriction to set duration as undefined if an empty string is provided, preventing API 400 errors. Updated related test to verify correct handling of empty duration values.

marinofranz commented on src/resources/universes.ts:
```diff
@@ -156,33 +155,47 @@ export class Universes {
    */
   async updateUserRestriction(
-    universeId: string,
     userRestrictionId: string,
```
Swap `userRestrictionId` with `universeId` and update the payload so it contains the `useRestrictionId` instead
This way, we stick to the code style of always specifying the `universeId` in the first parameter.

marinofranz commented on src/resources/universes.ts:
```diff
@@ -156,33 +155,47 @@ export class Universes {
+      console.log(payload);
+      payload.duration = undefined;
+      console.log(payload);
+    }
```
Refactor this to make it cleaner (remove the entire if statement)
```typescript
payload.duration = payload.duration === "" ? undefined : payload.duration;

```

marinofranz commented on src/resources/universes.ts:
```diff
@@ -156,33 +155,47 @@ export class Universes {
   ): Promise<UserRestriction> {
-    const searchParams = new URLSearchParams();
+    const { body, universeId, placeId } = options;
+    const payload = { ...body };
```
Is the shallow copy of `body` required here? I'm okay with this as long as it's got a reasonable explanation

commonly-ts replied:
oh I forgot to remove the stuff I was debugging the test with 💔

marinofranz commented on src/resources/universes.ts:
```diff
@@ -191,14 +204,20 @@ export class Universes {
+
+    // This resource has two different paths, one for universe-level restrictions, and another for place-level restrictions
+    if (placeId) {
+      resourcePath += `/places/${placeId}`;
+    }
+
+    resourcePath += `/user-restrictions/${userRestrictionId}`;
```
We can replace this with an array then removing any items that are falsey
```typescript
const resourcePath = [
  `/cloud/v2/universes/${universeId}`,
  placeId ? `/places/${placeId}` : null,
  `/user-restrictions/${userRestrictionId}`
].filter(Boolean).join("");
```

commonly-ts replied:
I do feel the if statement for the resourcePath is much more readable

marinofranz replied:
Fine with me, it might be changed in the future, it's undecided for now

marinofranz commented on src/resources/universes.ts:
```diff
@@ -191,14 +204,20 @@ export class Universes {
+
+    resourcePath += `/user-restrictions/${userRestrictionId}`;
+
+    return this.http.request<UserRestriction>(resourcePath.toString(), {
```
`.toString()` not required here since `resourcePath` is already a string

marinofranz reviewed (changes requested):
There are a few changes that I want to address regarding the code style, the logic is perfectly fine 🙂

commonly-ts commented:
also wondering about readability for `duration` as its not particularly clear that it needs to be in seconds with "s" at the end, unless you go and read the API docs.

could rename it to durationSeconds and have it as a number, and then parse it inside the function or is this SDK just a base wrapper without that sort of thing?

commonly-ts committed:
Refactor updateUserRestriction parameter order

Swapped the order of universeId and userRestrictionId in updateUserRestriction to match API expectations. Updated related type definitions, documentation, and tests for consistency.

marinofranz commented:
> also wondering about readability for `duration` as its not particularly clear that it needs to be in seconds with "s" at the end, unless you go and read the API docs.
> 
> could rename it to durationSeconds and have it as a number, and then parse it inside the function or is this SDK just a base wrapper without that sort of thing?

We try to stick to just the API reference but we can eventually figure out a helper function for that, this will do for now 🙂 
There's just one missing test that should test if `duration` is an empty string to pass coverage
